### ISSUE TITLE
B017: Not catching BaseException

### ISSUE THREAD

**Issue by saippuakauppias** [astral-sh/ruff#5462]
Example:
```python
from unittest import TestCase


class TestFoo(TestCase):
    def test_exception(self):
        with self.assertRaises(Exception):
            _ = 1

    def test_baseexception(self):
        with self.assertRaises(BaseException):
            _ = 1
```

Ruff output:
```shell
➜  ruff --select=B017 linters/ruff_errors/B017.py
linters/ruff_errors/B017.py:6:9: B017 `assertRaises(Exception)` should be considered evil
Found 1 error.
```

Not caught `BaseException`, although this class is the base class for all exceptions (including `Exception`): https://docs.python.org/3/library/exceptions.html#exception-hierarchy

ruff 0.0.270

**Comment by FishAlchemist** [comment]
### Although I don't know if it is correct not to catch BaseException, but flake8-bugbear seems to be not caught.
![image](https://github.com/astral-sh/ruff/assets/48265002/f1be5da0-d8a9-40ad-94f1-e2567c9676d2)

**Comment by charliermarsh** [comment]
I tend to agree that it should be included... Bugbear doesn't include it, and I can't find any documentation as to why. I will put up a PR and see if the ecosystem checks say flag this anywhere.

**Comment by saippuakauppias** [comment]
I don't know why it's not in other linters (maybe the developers forgot that `Exception` is not the root exception class), but I think it's very important and needed here.

Ruff should be better, smarter and more accurate than other linters. Adding `BaseException` class will make it a bit better.
